### PR TITLE
Add an `enabled` setting for all plugins

### LINKED ISSUE #3386: `enabled` setting for all plugins
It seems that so many plugins have an `enabled` setting these days that we may as well just add it in MkDocs. It will not run the plugin at all then. But if a plugin already has this setting explicitly, then MkDocs will not do anything, so the plugin can still decide what to do when it's disabled.

### PR DESCRIPTION
**Pull request by oprypin**
For every plugin, the user can set `enabled: false` (or something based on an environment variable) and it will not kick in at all.

If the plugin has its own `enabled` config, there's no change for it in that case.

* Closes #3386

### CODE REVIEW

**Review by ultrabug — APPROVED** [review]
Tested OK so far, cool

File: mkdocs/config/config_options.py [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ def load_plugin(self, name: str, config) -> plugins.BasePlugin:
         # Only if the plugin doesn't have its own "enabled" config, apply a generic one.
         if not any(pair[0] == 'enabled' for pair in plugin.config_scheme):
             if not config.get('enabled', True):
-                log.info(f"Plugin '{inst_name}' is disabled in the config, skipping.")
+                log.warning(f"Plugin '{inst_name}' is disabled in the config, skipping.")
```
Comment (ultrabug) [inline mkdocs/config/config_options.py:1144]: you meant `debug` right?

tbh I was fine with INFO but I guess DEBUG is more in line with how other plugins behave

Comment (oprypin) [reply mkdocs/config/config_options.py]: Oh what, yes I got things totally mixed up

**Review by ultrabug — APPROVED** [review]

**Review by pawamoy — APPROVED** [review]
Nicely done :+1: Exciting feature :slightly_smiling_face:

**Pull request merged**